testutil/pr/track.go

```diff
@@ -0,0 +1,289 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package pr provides functions to process GitHub pull requests.
+
+//nolint:wrapcheck,cyclop,exhaustruct
+package pr
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	gh "github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+const (
+	// Name of the GitHub organization.
+	organization = "ObolNetwork"
+	// The number of the project. For ex: https://github.com/orgs/ObolNetwork/projects/1 has projectNumber 1.
+	projectNumber = gh.Int(1)
+)
+
+// config represents the input fields used in graphql mutations.
+type config struct {
+	projectID       gh.ID
+	statusFieldID   gh.ID
+	doneOptionID    gh.String
+	sizeFieldID     gh.ID
+	sprintFieldID   gh.ID
+	currIterationID gh.ID
+}
+
+// Track tracks a PR without a ticket and adds it to the GitHub board.
+func Track(ghToken string) error {
```

Comment:
move all env checks out of this function and pass in as parameters to the function